sklearn/kernel_approximation.py

```diff
@@ -226,10 +227,7 @@ def transform(self, X):
 
         X = as_float_array(X, copy=True)
         X = check_array(X, copy=False)
-        if (X <= -self.skewedness).any():
-            raise ValueError("X may not contain entries smaller than"
-                             " -skewedness.")
-
+        check_non_negative(X, "SkewedChi2Sampler (input X)")
```

Comment:
Why did you replace the test? Now it will fail with things that previously worked.

Reply:
Sorry, I thought you meant to actually change the be behavior of the estimator (and indeed make it more restrictive) here: 
>Actually it's usually applied to positive data so maybe just add the tag asking for positivity, that should be fine.

Because the thing is AFAIU I can't just add a tag `requires_positive_X` but still let the estimator accept negative values (>-c), because `check_fit_non_negative` will then fail (a test than any estimator that has a `requires_positive_X` throws an error when given negative `X`) (actually it will fail because of the error message, but even if I change the error message into something like "Negative values in data passed to" (which I could tweak into sth like "Negative values in data passed to SkewedChi2Sampler after adding the constant c", which would be more exact) the test could still fail in principle because it could in principle test only on negative values that are >-c (it does not bc the negative values are random but logically it could)

Reply:
I agree with @amueller I would not change the behavior of the estimator to make the common tests pass. It just says that requires_positive_X is not what is adapted here.

Reply:
I agree, so should I create a new safe tag here ? Like `requires_X_higher_than_constant` or sth, that I could enforce in `enforce_estimator_tags_X` ? Because we need to find a way for tests that by default have negative values to translate them into `>c` for `SkewedChi2Sampler`

Reply:
Or maybe an `X_types` kind of tag ?